ffc/codegeneration/ufc.h

```diff
@@ -438,6 +438,15 @@ extern "C"
     ufc_tabulate_tensor_custom* tabulate_tensor;
   } ufc_custom_integral;
 
+  typedef struct ufc_expression
```

Comment:
The reference coords at which the expression is being evaluated, the number of points and the return shape should be provided as members.